pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -164,29 +182,26 @@ func (p Provisioner) Grant(options *apibkt.BucketOptions) (*bktv1alpha1.ObjectBu
 		return nil, errors.Wrapf(err, "bucket %s does not exist", p.bucketName)
 	}
 
-	// get or create ceph user
-	p.accessKeyID, p.secretAccessKey, err = p.createCephUser(options.UserID)
+	p.accessKeyID, p.secretAccessKey, err = bucket.getUserCreds()
 	if err != nil {
-		return nil, errors.Wrap(err, "Provision: can't create ceph user")
+		return nil, errors.Wrapf(err, "unable to get user %q creds", p.cephUserName)
 	}
 
-	// restrict creation of new buckets in rgw
-	restrictBucketCreation := 0
-	_, err = p.adminOpsClient.ModifyUser(p.clusterInfo.Context, admin.User{ID: p.cephUserName, MaxBuckets: &restrictBucketCreation})
-	if err != nil {
-		return nil, err
+	// do not modify externally managed users
+	if p.cephUserName == p.genUserName(options.ObjectBucketClaim) {
```

Comment:
Ditto concerns about `genUserName()` having changed over time

Reply:
This should be addressed by ` isObcGeneratedUser()`